### ISSUE TITLE
Sidebar shows Loading... -- no functionality at all

### ISSUE THREAD

**Issue by kklepper** [vctfence/scrapbee#13]
FF 66.0.3 64-bit

Step-3 | Finish Settings needs more explanation

```
first you have to realize there is a plus button

then the question is: what is meant by "name"??? directory name? 
should be (like c:\scrapbee)

and what about file? (test.rdf)

where is this data recorded to be able to check?

```
Restart Firefox

    ok, did that, to no avail

Enjoy

    I'd like to...
    RDF: box in sidebar is empty
    log file is fine

```
[info] ScrapBee version = 1.6.0
[info] browser = Firefox 66.0.3
[info] platform = Win32
[info] start web server on port 9900.
[info] start web server on port 9900.
[info] start web server on port 9900.
[info] start web server on port 9900.
[info] start web server on port 9900.
[info] start web server on port 9900.
[info] start web server on port 9900.
[info] web server started.
[info] web server started.
[info] web server started.
[info] web server started.
[info] web server started.
[info] web server started.
[info] web server started.
[info] show rdf list
[info] show rdf list
[info] show rdf list
[info] show rdf list
[info] show rdf list
[info] show rdf list
[info] show rdf list
[info] show rdf list

```

**Comment by vctfence** [comment]
Hi, the name is an alias of rdf file, this will be showed up in top-right drop down of sidebar instead of path of rdf file. So name is a free text, and path is full path of rdf file, please have a try. And I will make this more clear in later versions.

**Comment by joerngr** [comment]
Hi, first thanks for this great scrapbook replacement! 

Regarding the observed issue I just want to give a hint regarding an ugly pitfall I stumbled in:

I had some privacy settings – each time when my Firefox is closed, it deletes all history. Just website settings and offline website data are left. But removing cookies by that mechanism also removes the sqlite table content of ScrapBee. When I realized that, I started to test a bit. At "Cookies and website data" (sorry, I've a german UI active, maybe the actual translation is deviating) there is a possibility in the Firefox settings to take a look on cookies being set. For me that is - whyever - always flashvd.net, at least. When I remove all cokies from that list and wanna leave the dialog by "save changes", there is a message that sounds like "If you press this, you remove all cokie and website data". And, obviously, it does. Afterwards not only the cookies are gone, as well the local data from ScrapBee, all scrap directory settings. 

Not ydet completely clarified, but this Firefox behaviour looks strange, I obviously has to chose between keeping extensions like ScrapBee working OR getting rid of all those criminal sniffer cookies from all over the web. I hope, that this is a Firefox bug and not an intentional feature.

But, so far, it seems not to be an issue of ScrapBee.

**Comment by kklepper** [comment]
@joerngr Thank you, this is not my problem.
@vctfence Ok, this worked. 

So text is the name that gets displayed in the dropdown box, path the file to be opened. So I named my new thing `c:\scrapbee\test.rdf` "test" and my path to the old scrapbook file `C:\Users\BUDDY\AppData\Roaming\Mozilla\Firefox\Profiles\lrwtavtr.default\ScrapBook\scrapbook.rdf` "old" and now have 2 items to choose from. 

Great! All the old stuff is at hand... 

Easy to add more files, too. Thank you very much!

**Comment by kklepper** [comment]
Sorry, I have to reopen this case again. 

For your information:

As you will know, we had this Firefox certificate expiration issue. I found a workaround (see [wyxchari](https://borncity.com/win/2019/05/04/firefox-expired-certificate-disables-addons-may-4-2019/)). Later I removed that fix. After some time I noticed CPU usage 100% and scrapbee_backend.exe 34% in Process Explorer.

I disabled scrapbee -- nothing. I killed and restarted Firefox -- nothing. Log revealed: 

```
[info] ScrapBee version = 1.6.0
[info] browser = Firefox 66.0.3
[info] platform = Win32
[info] start web server on port 9900.
[error] listen tcp 127.0.0.1:9900: bind: Only one usage of each socket address (protocol/network address/port) is normally permitted.
[info] start web server on port 9900.
[error] listen tcp 127.0.0.1:9900: bind: Only one usage of each socket address (protocol/network address/port) is normally permitted.
[info] start web server on port 9900.
[error] listen tcp 127.0.0.1:9900: bind: Only one usage of each socket address (protocol/network address/port) is normally permitted.
[info] start web server on port 9900.
[error] listen tcp 127.0.0.1:9900: bind: Only one usage of each socket address (protocol/network address/port) is normally permitted.
[info] start web server on port 9900.
[error] listen tcp 127.0.0.1:9900: bind: Only one usage of each socket address (protocol/network address/port) is normally permitted.
[info] start web server on port 9900.
[error] listen tcp 127.0.0.1:9900: bind: Only one usage of each socket address (protocol/network address/port) is normally permitted.
[info] start web server on port 9900.
etc.
```

So I changed the port to 9001, but still the scrapbook dropdown box was not filled.

Due to the above mentioned problems, I killed Firefox and restarted; now the log reads

```
[info] ScrapBee version = 1.6.0
[info] browser = Firefox 66.0.3
[info] platform = Win32
[info] start web server on port 9901.
[info] web server started.
[info] show rdf list
[info] switch to rdf "C:\Users\WERNER\AppData\Roaming\Mozilla\Firefox\Profiles\lrwtavtr.default\ScrapBook\scrapbook.rdf"
[info] rdf loaded in 75ms
[info] switch to rdf "c:\scrapbee\test.rdf"
[info] rdf loaded in 3ms
```

The dropdown box is populated, switching works fine, but still CPU usage is 100% and scrapbee_backend.exe shows 32% in Process Explorer. 

What to do?

I looked at Process Explorer and found 3 instances of scrapbee_backend.exe, one of them normal, the others with high usage. I killed those, and I am fine again. Maybe you can kill leftover instances automatically when creating a new instance.

I notice some features I have to explore yet; this all looks very interesting.  Thank you once again.

**Comment by kklepper** [comment]
In order to investigate into those Firefox issues with old versions, I killed version 66 and loaded version 56 and back again -- now I had 2 scrapbee_backend.exe, one running full speed. So that is the reason behind this phenomenon.

**Comment by vctfence** [comment]
@kklepper Thank you very much for make these researching, Sometimes, we need to shutdown backend manually, for example, when firefox is not shuted down normally, then the backend is left running alone, By now, I am not sure if I can shutdown old process when new backend started, I am going to append some hints in help page about this.

**Comment by kklepper** [comment]
@vctfence Your software definitely deserves more attention. 

I wish you would explain more features. I just found out about marks on saved pages by trial and error. 

"Reload" cleared my marks, which is not what I expected. So what is "Mark Pen" for? Oh, I see, I have to save first. Ok, that's how it should be. Very good feature.

But what is "Page cleaning"? 

I expected my marks to be cleared, but instead -- after a while -- I got a cross cursor marking parts of my document. Clicking a highlighted part clears it -- so you can get rid of clutter. Great! Don't forget to stop and save.

Anything else I did not find out yet not mentioned in the help section?

**Comment by kklepper** [comment]
Maybe some background information is useful. 

Over time, I used quite a lot of tools to keep sketches and notices of my work, for example a wiki type tool named DokuWiki which was tedious to use with screenshots or generally parts of webpages, easy to do with OneNote, but after experiencing total OneNote data loss twice, I abandoned both and changed to Jupyter, which is great for documenting and running code, even embedding images like screenshots (a great advantage over DokuWiki), but not really good for documenting HTML code.

It seems that ScrapBee is the perfect companion to Jupyter, even easier to handle than OneNote, not to speak of constructions like Firefox Pocket which I never found any reason to use for real work.

I'm curious to find out if ScrapBee will redefine my workflow as I think it has the potential to do.

**Comment by kklepper** [comment]
Some feedback with the request for improvement.

In order to get my feet wet, I created a subfolder structure, which was straightforward. Next I performed a search which works fine.

In both cases I missed the possibility to confirm via Enter key instead of clicking the okay button. This would speed up these processes considerably.

**Comment by kklepper** [comment]
Another hint: opening a dialogue with an input field should set the cursor into this field instead of requiring the user to switch from keyboard to mouse and back.

**Comment by Kerenok** [comment]
As of 1.10.14, I experienced this bug : a sidebar blocked while Starting (no action is possible). 

Using the menu to go to Settings, I found that a new version was used 1.10.14. 

`[info] ScrapBee version = 1.10.4
[info] browser = Firefox 69.0.3
[info] platform = Win32
[info] start backend service on port 9900.
[info] backend service started, version = 1.7.0 (wanted >= 1.7.0)
[error] introNode.getAttributeNS(...) is null`

scrapbee_backend.exe seems to be launched correctly, no suspicious message in the browser's console.

In the future, could it be possible to revert to an old version of Scrapbee, for example, using this git repository?
